### PR TITLE
Add questions about optionals and state

### PR DESCRIPTION
**Pull request by diegopetrucci**

### CODE REVIEW

File: questions.md [hunk 1]
```diff
@@ -93,6 +94,8 @@ Please keep in mind that this document is in a continuous state of flux, with ne
 - Why is it that, in swift, you can't have an array of, say, `Equatable`, ie, what's the reason for the compiler error "Protocol 'XXX' can only be used as a generic constraint because it has Self or associated type requirements"?
 - Are you familiar with functional-programming methods such as `map`, `filter`, and `reduce`? Can you explain their similarities and differences?
 - `Swift Standard Library`: What are `Generator`s, `Sequence`s and `Collection`s?
+- What problem do optionals solve?
+- Do you use optionals to handle state? Errors? Why?
```
Comment (RuiAAPeres) [inline questions.md]: I would phrase it slightly differently, maybe:

`Do you use optionals to handle state? If so, how do you model Errors in an Optional context?

File: questions.md [hunk 2]
```diff
@@ -44,6 +44,7 @@ Please keep in mind that this document is in a continuous state of flux, with ne
 ```
 - What do you think are the main architectural challenges associated with developing for the iPad, compared to developing for the iPhone?
 - Do you have any experience developing for the Apple Watch or tvOS? If so, what's your opinion on the differences and similarities of the various development environments in the Apple development ecosystem?
+- How do you handle state in your app(s)?
```
Comment (wltrup) [inline questions.md:47]: That's a *very* open-ended question. How do you know if the answer is a good one or not? What kind of answers are you expecting here?

File: questions.md [hunk 3]
```diff
@@ -93,6 +94,8 @@ Please keep in mind that this document is in a continuous state of flux, with ne
 - Why is it that, in swift, you can't have an array of, say, `Equatable`, ie, what's the reason for the compiler error "Protocol 'XXX' can only be used as a generic constraint because it has Self or associated type requirements"?
 - Are you familiar with functional-programming methods such as `map`, `filter`, and `reduce`? Can you explain their similarities and differences?
 - `Swift Standard Library`: What are `Generator`s, `Sequence`s and `Collection`s?
+- What problem do optionals solve?
```
Comment (wltrup) [inline questions.md]: Great question! But I'd say `problems` (plural)

**Review by wltrup — CHANGES_REQUESTED** [review]

File: questions.md (hunk 2 above)
Comment (diegopetrucci) [reply questions.md:47]: I don't think there's just one good answer, it's mostly to see how the person thinks about this stuff.

Comment (wltrup) [reply questions.md:47]: > I don't there's just one answer

Either you mean "I don't. There's just one answer" or "I don't (know that) there's just one answer". I can't tell which.

**Review by wltrup — APPROVED** [review]
I'm approving but I still think that the state question is far too open-ended.

**Pull request merged**